### PR TITLE
Breaking changes

### PR DESCRIPTION
**Pull request by aurel-l**
Simplify a bit the scope of the component (stop listening for changes in sources in the subtree).

Apply changes mentioned in issue #1 

Some other cosmetic changes in the codebase

### CODE REVIEW

**Comment by ljgarcia** [comment]
I will take a look tomorrow afternoon. In the meanwhile, I am not sure if the <source ...> element is required in the detail response. We would have the src in any case so not sure what extra value the <source ...> srcElement adds. I am talking about the enhancements proposed at https://github.com/ebi-ppf/data-loader/issues/1

**Comment by aurel-l** [comment]
The main idea behind having both `src` and `srcElement` is for when you use a `<script type="application/json"…` instead of, or as a fallback to, a source element. But I guess you can deduce that if `src` is null.

**Comment by ljgarcia** [comment]
I see. In this case maybe better keeping both.

File: package.json [hunk 1]
```diff
@@ -30,25 +30,26 @@
...
+    "rimraf": "^2.6.1",
+    "rollup": "^0.43.0",
+    "rollup-plugin-babel": "^2.7.1",
     "rollup-plugin-eslint": "^3.0.0",
-    "rollup-plugin-node-resolve": "^2.1.0",
-    "rollup-watch": "^3.2.2",
-    "skatejs-web-components": "^5.0.0"
+    "rollup-plugin-node-resolve": "^3.0.0",
+    "rollup-watch": "^4.0.0",
+    "webcomponents.js": "github:webcomponents/webcomponentsjs#v1.0.1"
```
Comment (ljgarcia) [inline package.json]: Why do you need to this version and why pointing to github? I think the latest one is 1.02 and there is no need to point to github with that one.

File: src/data-loader.js [hunk 2]
```diff
@@ -1,59 +1,66 @@
...
+  async fetch () {
+    // get all the potentials sources elements
+    const sources = getSourceData(...this.children);
+    // if nothing there, bails
+    if (!sources.length) return;
 
-    let detail = [];
-    let failed = true;
-    for (const sourceDatum of sourceData) {
+    let errors = [];
```
Comment (ljgarcia) [inline src/data-loader.js]: indentation seems to be missing here

File: src/data-loader.js [hunk 3]
```diff
@@ -1,59 +1,66 @@
...
+  child => child.matches('source[src], script[type="application/json"]')
+);
 
 class DataLoader extends HTMLElement {
-  async _fetch () {
-    const sourceData = getSourceData(this);
-    if (!sourceData.length) return;
+  static get is () { return 'data-loader' }
+
+  async fetch () {
```
Comment (ljgarcia) [inline src/data-loader.js]: This fetch is kind of long, maybe the code corresponding to the loops can be taken to private methods.

File: src/data-loader.js (hunk 2 above)
Comment (aurel-l) [inline src/data-loader.js]: Hummm, no, it looks alright

File: src/data-loader.js (hunk 3 above)
Comment (aurel-l) [inline src/data-loader.js]: Agree, I extracted the inside of the loop into a fetchOne function

File: package.json (hunk 1 above)
Comment (aurel-l) [inline package.json]: The latest one publish, to date, seems to be 1.0.1, I didn't see someone published it to npm, I will use the one from npm then

**Comment by aurel-l** [comment]
Updated after request for changes was made

**Pull request merged**